.github/workflows/type-safety.yml

```
@@ -0,0 +1,89 @@
+    types: [opened, synchronize, reopened]
+
+jobs:
+  type-safety-check:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      
+    - name: Setup Go
+      uses: actions/setup-go@v4
```

actionlint (1.7.8)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)